File: src/main/kotlin/com/github/max/api/navigator/reference/XmlApiReference.java
package com.github.max.api.navigator.reference;

import com.intellij.codeInsight.lookup.LookupElementBuilder;
import com.intellij.psi.*;
import com.intellij.psi.xml.XmlAttributeValue;
import com.intellij.util.PlatformIcons;
import org.apache.commons.lang.StringUtils;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import java.util.Arrays;

public class XmlApiReference extends PsiReferenceBase<XmlAttributeValue> {

    private final PsiClass psiClass;

    public XmlApiReference(@NotNull XmlAttributeValue element, PsiClass psiClass) {
        super(element);
        this.psiClass = psiClass;
    }


    @Nullable
    @Override
    public PsiElement resolve() {
        if (!StringUtils.isEmpty(myElement.getValue())) {
            PsiMethod[] methods = psiClass.findMethodsByName(myElement.getValue(), false);
            return methods.length == 0 ? null : methods[0];
        }
        return null;
    }

    @NotNull
    @Override
    public Object[] getVariants() {
        return Arrays.stream(psiClass.getMethods())
                .map(method -> {
                            String[] signs = Arrays.stream(method.getParameterList().getParameters())
                                    .map(PsiParameter::getType)
                                    .map(PsiType::getPresentableText)
                                    .toArray(String[]::new);
                            String sign = StringUtils.join(signs, ", ");
                            return LookupElementBuilder.create(method)
                                    .withIcon(PlatformIcons.METHOD_ICON)
                                    .withTailText("(" + sign + ")");
                        }
                ).toArray();
    }
}

File: src/main/kotlin/com/github/max/api/navigator/reference/XmlApiReference.kt
package com.github.max.api.navigator.reference

import com.intellij.codeInsight.lookup.LookupElementBuilder
import com.intellij.psi.*
import com.intellij.psi.xml.XmlAttributeValue
import com.intellij.util.PlatformIcons
import org.apache.commons.lang.StringUtils
import java.util.*

class XmlApiReference(element: XmlAttributeValue, private val psiClass: PsiClass)
    : PsiReferenceBase<XmlAttributeValue?>(element) {

    override fun resolve(): PsiElement? {
        if (!StringUtils.isEmpty(myElement!!.value)) {
            val methods = psiClass.findMethodsByName(myElement.value, false)
            return if (methods.isEmpty()) null else methods[0]
        }
        return null
    }

    override fun getVariants(): Array<Any> {
        return Arrays.stream(psiClass.methods)
                .map { method: PsiMethod ->
                    val signs = Arrays.stream(method.parameterList.parameters)
                            .map { obj: PsiParameter -> obj.type }
                            .map { obj: PsiType -> obj.presentableText }
                            .toArray()
                    val sign = StringUtils.join(signs, ", ")
                    LookupElementBuilder.create(method)
                            .withIcon(PlatformIcons.METHOD_ICON)
                            .withTailText("($sign)")
                }.toArray()
    }
}